src/exec_context.ts

```diff
@@ -198,11 +215,29 @@ export default class ExecContext<M> {
       return fn.call(this, ...args);
     };
 
+    const mergeContainerEnv = (parent?: ExecContext<M>, env?: Environment): Environment => {
```

Comment:
I wanted to move this out of `ExecContext` but types were yelling at me because of `parent?: ExecContext<M>`

Reply:
Yeah, the type parameter for the model has to pass through everything that touches it.